public/app/plugins/datasource/prometheus/configuration/overhaul/ConnectionSettings.tsx

```diff
@@ -0,0 +1,90 @@
+import { css, cx } from '@emotion/css';
+import React, { ReactNode } from 'react';
+
+import { DataSourceJsonData, DataSourceSettings } from '@grafana/data';
+import { ConfigSection } from '@grafana/experimental';
+import { InlineField, Input, PopoverContent } from '@grafana/ui';
+
+import { PromOptions } from '../../types';
+// THIS FILE IS COPIED FROM GRAFANA/EXPERIMENTAL
+// BECAUSE IT CONTAINS TYPES THAT ARE REQUIRED IN THE ADVANCEDHTTPSETTINGS COMPONENT
+// THE TYPES ARE WRITTEN IN EXPERIMENTAL WHERE THEY ARE NOT AS STRICT
+// @ts-ignore
+export type Config<JSONData extends DataSourceJsonData, SecureJSONData> = DataSourceSettings<
+  // @ts-ignore
+  JSONData,
+  // @ts-ignore
+  SecureJSONData
+>;
+// @ts-ignore
+export type OnChangeHandler<C extends Config = Config> = (options: DataSourceSettings<PromOptions, {}>) => void;
+// @ts-ignore
```

Comment:
Do we know when we clean those `@ts-ignore` lines?
Also I think it might be a good idea to have an issue to track this.

Reply:
I have added it to the grooming of the config in this issue
https://github.com/grafana/grafana/issues/72385

Reply:
We can clean them up once we change the experimental components style in grafana/experimental